client/my-sites/checkout/composite-checkout/lib/update-cart-contact-details-for-checkout.ts

```diff
@@ -43,5 +43,7 @@ export async function updateCartContactDetailsForCheckout(
 		vatId: vatDetails.id ?? '',
 		organization: vatDetails.name ?? '',
```

Comment:
I think this will need to change slightly to properly support `organization`. Currently it's assuming that the organization comes from the VAT details endpoint, but with this change we might instead want to use the contact details version of the value. 🤔  But how do we know when to choose? We don't want to use the contact details organization value if the VAT details are active or if the organization field is not displayed (not required by the country).

Reply:
Right now, it would only show the organization in contact details if both are present.

Reply:
* If domain contact details is being used, organization is excluded from VAT details
* If an organization is required for tax, organization is excluded from VAT details
* If an organization is not required for tax, it's set to undefined in `contactInfo`
* I think this means we can default to `contactInfo.organization` and fall back to `vatDetails.name` as long as `vatDetails.name` is unset if that form is hidden

Reply:
> I think this means we can default to contactInfo.organization and fall back to vatDetails.name as long as vatDetails.name is unset if that form is hidden

I think you're right since in this case `vatDetails` is the internal `wpcom-checkout` data store version of the value. However, you can prove it. You can write tests for the different scenarios by adding to these:

https://github.com/Automattic/wp-calypso/blob/feab57902d951c3e9b84e997f4a76def4bdaca94/client/my-sites/checkout/composite-checkout/test/checkout-contact-step.js#L706